automation/roles/tls_certificate/generate/tasks/main.yml

```diff
@@ -0,0 +1,122 @@
+---
+- name: Make sure that the python3-cryptography package is present
+  ansible.builtin.package:
+    name: python3-cryptography
+    state: present
+  register: pack_status
+  until: pack_status is success
+  delay: 5
+  retries: 3
+
+- name: "Clean up existing certificates"
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: absent
+  loop:
+    - "{{ tls_privatekey_path | default('/etc/tls/server.key') }}"
+    - "{{ tls_cert_path | default('/etc/tls/server.crt') }}"
+    - "{{ tls_ca_cert_path | default('/etc/tls/ca.crt') }}"
+    - "{{ tls_ca_privatekey_path | default('/etc/tls/ca.key') }}"
+    - "{{ tls_etcd_cert_path | default('/etc/etcd/server.crt') }}"
+    - "{{ tls_etcd_ca_cert_path | default('/etc/etcd/ca.crt') }}"
+    - "{{ tls_etcd_privatekey_path | default('/etc/etcd/server.key') }}"
+    - "/etc/tls"
```

Comment:
```suggestion
```
Let’s avoid deleting the entire directory. To minimize the risk of accidentally deleting certificates unrelated to the PostgreSQL cluster, it makes sense to handle this more cautiously when converting an existing installation into a cluster. In such cases, we cannot assume in advance whether there are other certificates in the directory besides the ones we generate.

Reply:
Or perhaps we could remove this task entirely. Could you clarify the purpose of deleting the existing certificates, if any? It would help us understand if this step is necessary or if there’s an alternative approach.

Reply:
The `/etc/tls` directory does not exist by default on any Debian- or RHEL-based systems. Therefore, the only files present there will be the ones we explicitly add. In this case, it’s just the certificate for the PostgreSQL cluster. This means it should safe to delete if needed.

The reason I included that task is to ensure flexibility for re-initializing the cluster. For example, if you want to reinitialize it with different hostnames or IPs, the etcd cluster will not function correctly. This is because the IPs and hostnames must be part of the certificate's Subject Alternative Name (SAN). Without this, encrypted communication between etcd nodes would fail. It will complain with messages like `Bad Certificate`

Additionally, if you plan to add new nodes to an existing cluster in the future (thereby expanding the etcd cluster), those certificates will also need to be regenerated.

Reply:
However, since we are only deleting specific files, we should not delete the entire ‘tls’ directory.